usaspending_api/recipient/v2/views.py

```diff
@@ -1,94 +1,168 @@
 import logging
+from collections import OrderedDict
 from datetime import datetime
 from dateutil.relativedelta import relativedelta
-from collections import OrderedDict
-
-from rest_framework.response import Response
-from usaspending_api.common.views import APIDocumentationView
 
+from django.conf import settings
+from django.contrib.postgres.aggregates import StringAgg
 from django.db.models import Sum
+from rest_framework.response import Response
 
-from usaspending_api.awards.v2.filters.filter_helpers import sum_transaction_amount
-# from usaspending_api.awards.v2.filters.sub_award import subaward_filter
-from usaspending_api.awards.v2.filters.matview_filters import universal_transaction_matview_filter
+from usaspending_api.awards.models_matviews import SummaryStateView
+from usaspending_api.awards.v2.filters.matview_filters import matview_search_filter
+from usaspending_api.awards.v2.lookups.lookups import all_award_types_mappings as ats
 from usaspending_api.common.exceptions import InvalidParameterException
 from usaspending_api.common.helpers.generic_helper import generate_fiscal_year
+from usaspending_api.common.views import APIDocumentationView
 from usaspending_api.recipient.models import StateData
 
 logger = logging.getLogger(__name__)
 
+# Storing FIPS codes + state codes in memory to avoid hitting the database for the same data
+VALID_FIPS = {}
 
-class StateMetaDataViewSet(APIDocumentationView):
 
-    def get_state_data(self, state_data_results, field, year=None):
-        """Finds which earliest or latest state data to use based on the year and what data is available"""
-        state_data = OrderedDict(sorted([(str(state_data['year']), state_data) for state_data in state_data_results
-                                         if state_data[field]], key=lambda pair: pair[0]))
-        earliest = list(state_data.keys())[0]
-        latest = list(state_data.keys())[-1]
-        if year and year.isdigit() and year < earliest:
-            return state_data[earliest]
-        elif year and year.isdigit() and earliest <= year <= latest:
-            return state_data[year]
-        elif not year or year > latest or year == 'all' or year == 'latest':
-            return state_data[latest]
-        else:
-            raise InvalidParameterException('Invalid year: {}.'.format(year))
+def populate_fips():
```

Comment:
Should this function live in this file if it's specific to state profile and this `views.py` file is generic to all recipient viewsets?

Reply:
`views.py` has been moved into `recipient/v2/views/states.py` to remove the general nature of the file.